### PR TITLE
Testing components

### LINKED ISSUE #22: Testing: NavBar Testing
As a developer I want to test the NavBar Component

### LINKED ISSUE #23: Testing: Form Testing
As a developer I want to test the Form Component

### LINKED ISSUE #24: Testing: FindARoommate Testing
As a developer I want to test the FindARoommate component

### PR DESCRIPTION
**Pull request by chadrick-d-dev**
### What’s this PR do?
* Creates passing tests for NavBar, Form, and FindARoommate
* Testing for NavBar and Form are complete
* Testing for FindARoommate requires more tests
* Updates gitignore by adding eslintcache to ignored files

### Where should the reviewer start?
* Review Test files

### How should this be manually tested?
* Run ```npm install``` for dependencies.
* Run ```npm test``` to run tests.

### Any background context you want to provide?
* As developers we need to run unit, integration, and async tests on our components, depending on what kind of testing they need.

### CODE REVIEW

File: src/NavBar/NavBar.test.js [hunk 1]
```diff
@@ -1,20 +1,55 @@
...
+  
+    const aboutTab = screen.getByText('About')
+    const findRoommateTab = screen.getByText('Find A Roommate')
+  
+    expect(aboutTab).toBeInTheDocument()
+    expect(findRoommateTab).toBeInTheDocument()
+  })
+
+  it('should route user to About page when About tab is clicked', async () => {
+    const history = createMemoryHistory()
```
Comment (cameronRomo) [inline src/NavBar/NavBar.test.js:25]: Super handy method.

File: src/FindARoommate/FindARoommate.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  it('should render list with resident thumbnails', async () => {
+    // setup and execution 
+    render(
+      <MemoryRouter>
+        <FindARoommate/>
+      </MemoryRouter>
+    )
+    //execution
+    const thumbnailsList = screen.getByRole('list')
+    const dorisName = await waitFor(() => screen.getByText('Doris'))
```
Comment (cameronRomo) [inline src/FindARoommate/FindARoommate.test.js:50]: Cool how you can just do one waitFor for loading the api. Totally makes sense.

File: src/FindARoommate/FindARoommate.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ export default function FindARoomate() {
   const [semester, setSemester] = useState('');
   const [residents, setResidents] = useState([]);
   const [availableResidents, setAvailableResidents] = useState([]);
-  const [selectedResident, setSelectedResident] = useState('');
+  const [selectedResident, setSelectedResident] = useState(null);
```
Comment (lo-la-do-li) [inline src/FindARoommate/FindARoommate.js:35]: Good catch - Null makes more sense!

**Review by cameronRomo — APPROVED** [review]
There are some really strong tests written here!

File: src/FindARoommate/FindARoommate.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  })
+
+  it('should render Form', () => {
+    render(
+      <MemoryRouter>
+        <FindARoommate/>
+      </MemoryRouter>
+    )
+    
+    const semesterDropdownForm = screen.getByRole('button', { name: /choose a semester ​/i })
```
Comment (lo-la-do-li) [inline src/FindARoommate/FindARoommate.test.js:36]: Learned something new with the regular expression on getByRole. Nice work!

File: src/NavBar/NavBar.test.js [hunk 5]
```diff
@@ -1,20 +1,55 @@
...
 import userEvent from '@testing-library/user-event';
+import { createMemoryHistory } from "history";
 
 describe('NavBar', () => {
-  //This test file will mostly be for unit testing
-  it.skip('It should display a description of the app and the collaboration', () => {
-    // setup and execution 
+  it('should render navbar with About and Find A Roommate', () => {
     render(
+      <MemoryRouter>
```
Comment (lo-la-do-li) [inline src/NavBar/NavBar.test.js:12]: So glad you figured this one out!

**Review by lo-la-do-li — APPROVED** [review]

**Pull request merged**